ibis/backends/dask/aggcontext.py

```diff
@@ -0,0 +1,235 @@
+        if isinstance(function, str):
+            return getattr(grouped_data, function)(*args, **kwargs)
+
+        if not callable(function):
+            raise TypeError(f'Object {function} is not callable or a string')
+
+        if isinstance(grouped_data, SeriesGroupBy) and len(grouped_data):
+            # `SeriesGroupBy.agg` does not allow np.arrays to be returned
+            # from UDFs. To avoid `SeriesGroupBy.agg`, we will call the
+            # aggregation function manually on each group. (#2768)
+            aggs = {}
+            for k, v in grouped_data:
+                func_args = [d.get_group(k) for d in args]
+                aggs[k] = function(v, *func_args, **kwargs)
+                grouped_col_name = v.name
+            return (
+                dd.Series(aggs)
+                .rename(grouped_col_name)
+                .rename_axis(grouped_data.grouper.names)
+            )
+        elif isinstance(grouped_data, dd.Series):
+            return grouped_data.reduction(wrap_for_agg(function, args, kwargs))
+        else:
+            return grouped_data.agg(wrap_for_agg(function, args, kwargs))
+
+
+class Transform(AggregationContext):
+    __slots__ = ()
+
+    def agg(self, grouped_data, function, *args, **kwargs):
+        # If this is a multi column UDF, then we cannot use
+        # "transform" here (Data must be 1-dimensional)
+        # Instead, we need to use "apply", which can return a non
+        # numeric type, e.g, tuple of two double.
+        if isinstance(self.output_type, dt.Struct):
+            res = grouped_data.apply(function, *args, **kwargs)
+        else:
+            res = grouped_data.transform(function, *args, **kwargs)
+        # The result series uses the name of the input. We should
+        # unset it to avoid confusion, when result is not guranteed
+        # to be the same series / have the same type after transform
+        res.name = None
+        if res.index.name is not None:
+            index_name = res.index.name
+            # reset_index turns series to dataframe, need to keep it as a series.
+            res = res.reset_index().set_index(index_name).iloc[:, 0]
+
+        return res
+
+
+def dask_window_agg_built_in(
+    frame: dd.DataFrame,
+    windowed: dd.rolling.Rolling,
+    function: str,
+    max_lookback: int,
+    *args: Tuple[Any],
+    **kwargs: Dict[str, Any],
+) -> dd.Series:
+    """Apply window aggregation with built-in aggregators."""
+    assert isinstance(function, str)
+    method = operator.methodcaller(function, *args, **kwargs)
+
+    if max_lookback is not None:
+        agg_method = method
+
+        def sliced_agg(s):
+            return agg_method(s.iloc[-max_lookback:])
+
+        method = operator.methodcaller('apply', sliced_agg, raw=False)
+
+    result = method(windowed)
+    # No MultiIndex support in dask
+    result.index = frame.index
+    return result
+
+
+class Window(AggregationContext):
```

Comment:
Most of this is a direct-ish port of what's in the pandas backend?

Reply:
Yeah, it's pretty much just a Dask port from the Pandas backend version.
The difference is mostly just needing to make sure the resulting dataframes are partitioned for Dask operations downstream and where the Dask api is slightly different than Pandas api.